### PR TITLE
sci-misc/boinc: add 7.18.1

### PR DESCRIPTION
**Pull request by Kangie**
Signed-off-by: Matt Jolly <Matt.Jolly@footclan.ninja>

### CODE REVIEW

**Comment by Kangie** [comment]
As an aside, we should look at adding a `systemd` use flag and a unit file for boinc; the openrc ones don't work ootb.

The [Arch package](https://archlinux.org/packages/community/x86_64/boinc/) has a unit file that worked for me to do some quick testing of the package; I might see about adding something to the ebuild tomorrow.

**Comment by thesamesam** [comment]
I'll handle it here but remember to sync live ebuilds if they exist

**Comment by thesamesam** [comment]
> As an aside, we should look at adding a `systemd` use flag and a unit file for boinc; the openrc ones don't work ootb.
> 
> The [Arch package](https://archlinux.org/packages/community/x86_64/boinc/) has a unit file that worked for me to do some quick testing of the package; I might see about adding something to the ebuild tomorrow.

ack I'll hold off on merging

**Comment by Kangie** [comment]
> > As an aside, we should look at adding a `systemd` use flag and a unit file for boinc; the openrc ones don't work ootb.
> > The [Arch package](https://archlinux.org/packages/community/x86_64/boinc/) has a unit file that worked for me to do some quick testing of the package; I might see about adding something to the ebuild tomorrow.
> 
> ack I'll hold off on merging

So, turns out upstream ships a working unit file and I don't know what I was thinking.

On the plus side, -9999 has been updated & tested and the ebuild has been tidied up overall.

There's a patch that we apply from [2009](https://bugs.gentoo.org/286701) that we might be able to do away with... know anyone with a PPC?

Should be good to merge now.

**Comment by thesamesam** [comment]
> 

yes! But I can't test it right this moment and don't want to put off merging this longer. But next time.

**Comment by thesamesam** [comment]
Duplicate patch:
```
$ pkgcheck scan
sci-misc/boinc
  DuplicateFiles: duplicate identical files in FILESDIR: 'files/boinc-7.18-fix_webview.patch', 'files/fix_webview.patch'
  UnstableOnly: for arches: [ amd64, arm64, ppc, ppc64, sparc, x86 ], all versions are unstable: [ 7.18.1 ]
  DeprecatedDep: version 7.18.1: DEPEND: deprecated dependencies: virtual/jpeg virtual/jpeg:0=
  DeprecatedDep: version 7.18.1: RDEPEND: deprecated dependencies: virtual/jpeg virtual/jpeg:0=
  UnusedInherits: version 7.18.1: unused eclass: systemd
  DeprecatedDep: version 9999: DEPEND: deprecated dependencies: virtual/jpeg virtual/jpeg:0=
  DeprecatedDep: version 9999: RDEPEND: deprecated dependencies: virtual/jpeg virtual/jpeg:0=
  UnusedInherits: version 9999: unused eclass: systemd
```

Also, is the PPC patch.. gone now?

**Comment by Kangie** [comment]
Duplicate patch is how this ebuild was when I found it - I've just tidied up usage. There used to be fix_webview and ver-fix_webview and live and latest pointed at the same ver. 

Now live points at the generic patch (which I updated) so that it can act as a canary and be updated for the next release if anyone is using live.

Ppc patch is applied with sed in src-prepare.

Sorry about formatting; on mobile.

**Comment by thesamesam** [comment]
Oh right, I see. I'd just have live pointing at the last one you rebased. If the patches are identical, we don't want both sitting in tree, as it's a waste of space.

**Comment by Kangie** [comment]
Can do, give it an hour or so and check back.

File: sci-misc/boinc/boinc-9999.ebuild [hunk 1]
```diff
@@ -29,7 +29,7 @@ REQUIRED_USE="^^ ( curl_ssl_gnutls curl_ssl_openssl ) "
 
 # libcurl must not be using an ssl backend boinc does not support.
 # If the libcurl ssl backend changes, boinc should be recompiled.
-COMMON_DEPEND="
+DEPEND="
 	acct-group/boinc
```
Comment (Flowdalic) [inline sci-misc/boinc/boinc-9999.ebuild:33]: I am not sure if we have a policy on this, but it is unnecessary to explicitly state the dependency on the user's `acct-group` package, as `acct-user/boinc` already depends on `acct-group/boinc`. Maybe a good opportunity to fix in the live ebuild and the revbumped ebuild.

Comment (Kangie) [reply sci-misc/boinc/boinc-9999.ebuild:33]: Hasn't been merged yet and I'm at my dev PC...

Comment (Flowdalic) [reply sci-misc/boinc/boinc-9999.ebuild:33]: This was just briefly discussed in #gentoo-qa, there doesn't appear to be a policy nor consensus about this. So feel free to keep as is.

**Comment by Flowdalic** [comment]
Thanks for your contribution. Note that I did some follow-up cleanups in https://github.com/gentoo/gentoo/commit/e172239312f312f1385cfb00635c85ab89fef1e8 that `pkgcheck scan` revealed. You may want to look into the `pkgcheck` tool. :)

**Comment by juippis** [comment]
Why is there 3 separate commits for a new version, 7.18.1? These should've been combined to a single one. Why doesn't it list all required deps?

@Flowdalic please rebase on your own if the contributor isn't too familiar using git.

**Comment by Flowdalic** [comment]
> Why doesn't it list all required deps?

It listed all the required dependencies. You re-added the unnecessary acct-group ones. And as we had established in #gentoo-qa, there is no policy on that.

> @Flowdalic please rebase on your own if the contributor isn't too familiar using git.

Will do.

**Comment by juippis** [comment]
> 
> It listed all the required dependencies. You re-added the unnecessary acct-group one. And as we had established in #gentoo-qa, there is no policy on that.
> 

**If** the running program requires this specific group, then it's needed right? If it doesn't, then why such package exists? I did consider the possibility of upstream dropping this runtime requirement, but never seen it happen in reality.

**Comment by Flowdalic** [comment]
I am not arguing that they system group is not required, I am arguing that explicitly stating it as dependency is unnecessary as the `acct-user/boinc` will always depend on `acct-group/boinc` and thus the group will always be available at runtime. Now, the situation would be different if the boinc ebuild did some action that required the existence of the 'boinc' system group, as `acct-user` packages only declare a RDEPEND on their specified groups. Then you have to explicitly declare the according DEPEND. This is, however, not the case in the current boinc ebuild.

**Comment by thesamesam** [comment]
There's no obligation for a user to have a corresponding group and it's an implementation detail. If this package at runtime uses the group too, you should depend on it.

Dependencies of other packages should be considered a black box and not introspectable.

**Comment by juippis** [comment]
Btw sorry if my tone came out harsh here (internet). Thank you both, appreciate work that benefits Gentoo.

@Flowdalic see https://github.com/gentoo/devmanual/pull/284 - maybe we can make it a policy!

**Comment by Flowdalic** [comment]
Sure, that's just a friendly technical discussion of people wo are not in agreement.

FWIW, and I don't want to come over as overly picky, but it appears that boinc does not require the 'boinc' system group. The `setgid()`s I found in boinc's source seem to be unrelated, at least they do not appear to expect the presence of a 'boinc' system group, and finally, the `boinc-client.service` has no `Group=` directive. BTW, the latter is IMHO a good thing, far to many systemd .service files unnecessarily specify a system group for the service, systemd's fallback to the user's primary group is what you typically want here instead.

So where exactly does boinc require this specific group?

**Comment by juippis** [comment]
> FWIW, and I don't want to come over as overly picky, but it appears that boinc does not require the 'boinc' system group. The `setgid()`s I found in boinc's source seem to be unrelated, at least they do not appear to expect the presence of a 'boinc' system group, and finally, the `boinc-client.service` has no `Group=` directive. BTW, the latter is IMHO a good thing, far to many systemd .service files unnecessarily specify a system group for the service, systemd's fallback to the user's primary group is what you typically want here instead.

That's also what I asked previously :)

> 
> So where exactly does boinc require this specific group?

At least in the openrc init script it seems to. Not sure if that's necessary though, but I wonder if we can revert that now. It really could be the group was originally added "just in case". (although, the init.d executable is pretty big, it probably come from upstream?)

**Pull request closed without merging**